src/imputation/MoR.py

```diff
@@ -63,17 +71,21 @@ def mor_preprocessing(df, backdata):
     df["cf_group_size"] = np.nan
 
     # TODO move this to imputation main
-    # Select only values to be imputed
+    # Create imp_class column
     df = create_imp_class_col(df, "200", "201")
 
-    imputation_cond = (df["formtype"] == "0001") & (df["status"].isin(bad_statuses))
+    stat_cond = df["status"].isin(bad_statuses)
+    sf_cond = (df["formtype"] == "0006") & (df["selectiontype"] == "C")
+    lf_cond = df["formtype"] == "0001"
+    imputation_cond = stat_cond & (sf_cond | lf_cond)
     to_impute_df = df.copy().loc[imputation_cond, :]
     remainder_df = df.copy().loc[~imputation_cond, :]
 
-    clear_status_cond = backdata["status"].isin(good_statuses)
+    # Ensure backdata is as we require
+    wanted_cond = backdata["imp_marker"].isin(["R", "CF", "MoR", "TMI"])
+    backdata = backdata.copy().loc[wanted_cond, :]
```

Comment:
loc[wanted_cond] - I think it would work without the colon, but it's not important